pkg/operator/staticpod/startupmonitor/monitor.go

```diff
@@ -106,18 +106,22 @@ func (m *monitor) Run(ctx context.Context, installerLock Locker) (ready bool, re
 	// we use wait.Until because it uses sliding interval, wait.Poll does not.
 	withTimeoutCtx, cancel := context.WithTimeout(ctx, m.timeout-m.probeInterval) // last iteration is done manually to have control over the Unlock
 	wait.UntilWithContext(withTimeoutCtx, func(ctx context.Context) {
-		defer func() {
-			if lastReady {
-				installerLock.Unlock()
-			}
-		}()
+		defer installerLock.Unlock()
```

Comment:
no, this breaks the contract.

Reply:
Only on error or timeout we are allowed to release the lock. If the target is ready here in the loop, we have to keep the lock kept.

Reply:
fixed.